SessionMessagingKit/Jobs/Types/GroupLeavingJob.swift

```diff
@@ -0,0 +1,167 @@
+// Copyright © 2023 Rangeproof Pty Ltd. All rights reserved.
+
+import Foundation
+import GRDB
+import PromiseKit
+import SignalCoreKit
+import SessionUtilitiesKit
+import SessionSnodeKit
+
+public enum GroupLeavingJob: JobExecutor {
+    public static var maxFailureCount: Int = 0
+    public static var requiresThreadId: Bool = true
+    public static var requiresInteractionId: Bool = true
+    
+    public static func run(
+        _ job: SessionUtilitiesKit.Job,
+        queue: DispatchQueue,
+        success: @escaping (SessionUtilitiesKit.Job, Bool) -> (),
+        failure: @escaping (SessionUtilitiesKit.Job, Error?, Bool) -> (),
+        deferred: @escaping (SessionUtilitiesKit.Job) -> ())
+    {
+        print("Ryan Test: Group leaving job running")
+        guard
+            let detailsData: Data = job.details,
+            let details: Details = try? JSONDecoder().decode(Details.self, from: detailsData),
+            let interactionId: Int64 = job.interactionId
+        else {
+            failure(job, JobRunnerError.missingRequiredDetails, false)
+            return
+        }
+        
+        guard let thread: SessionThread = Storage.shared.read({ db in try? SessionThread.fetchOne(db, id: details.groupPublicKey)}) else {
+            SNLog("Can't leave nonexistent closed group.")
+            failure(job, MessageSenderError.noThread, false)
```

Comment:
Should this fail or succeed? (if the group doesn't exist then it's already left?)

If you do want it to fail then you should probably pass `true` for the last parameter (which indicates a permanent failure)

Reply:
Hmm... A permanent fail might make more sense to me. I'll change it.